src/views/virtualmachines/actions/hooks/useMultipleVirtualMachineActions.ts

```diff
@@ -0,0 +1,72 @@
+import { useMemo } from 'react';
+import { VirtualMachineModel } from 'src/views/dashboard-extensions/utils';
+
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { ActionDropdownItemType } from '@kubevirt-utils/components/ActionsDropdown/constants';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { getNamespace } from '@kubevirt-utils/resources/shared';
+import { K8sVerb } from '@openshift-console/dynamic-plugin-sdk';
+
+import { restartVM, startVM, stopVM, unpauseVM } from '../actions';
+
+type UseMultipleVirtualMachineActions = (vms: V1VirtualMachine[]) => ActionDropdownItemType[];
+
+const useMultipleVirtualMachineActions: UseMultipleVirtualMachineActions = (vms) => {
+  const { t } = useKubevirtTranslation();
+  const namespaces = Array.from(new Set(vms?.map((vm) => getNamespace(vm))));
+
+  const namespace = namespaces.length > 1 ? undefined : namespaces[0];
```

Comment:
you can bind both in useMemo and no need for Array.from , set has size property

Reply:
to get first element if u need u can use
const [first] = set;

Reply:
i would remove the accessReview. Thinking more about that. 
If the user selects vms into two different namespaces, we will ask if he has access to all namespaces. 
If the user is not admin it will not be able to perform actions in this case so I would make the action already available. If the user does not have permissions to perform the action, it will just not happen

Reply:
Example:
I have two vms into 'default' and 'my-test-namespace', I select both vms and try to make an action.
The UI should not verify if I have access to all namespaces. It should verify that I have access to only those two namespaces. This means that we have to make two different fetch calls. One to verify that I can patch vms in the 'default' namespace and one for 'my-test-namespace'.  This is not scalable for multiple namespaces in case of a user that select all vms in all available namespaces.

Reply:
I will just not verify anything at this point. If the user does not have permissions its action will not do anything